subprojects/docs/src/docs/userguide/authoringMaintainableBuildScripts.adoc

```diff
@@ -47,8 +101,8 @@ Documentation tasks
 generateDocs - Generates the HTML documentation for this project.
 ----
 
-[[sec:minimize_logic_executed_configuration_phase]]
-=== Minimize logic executed during the configuration phase
+[[sec:avoiding_execution_actions_during_configuration]]
+=== Avoiding execution actions during Gradle configuration
```

Comment:
I personally find this title a little confusing. There's no term like "execution actions". We should try to find a better title here IMHO if you think it's not clear.

Reply:
Good call

Reply:
How about "Avoiding I/O during Gradle configuration"?

Reply:
I don't think this is really about I/O. It's about any unnecessary code being executed during the configuration phase. To be honest I don't mind my initial proposal.

Here're some other ideas. They don't follow the pattern of do and don't.

- Logic outside of task actions will always execute
- Identifying unnecessary configuration code
- Reducing the cost on performance for configuring a build

Reply:
My interpretation of this subtopic is that it's all about performance. My issue is that one could have lots of "logic" and configuration still be fast. 

How about "Avoiding expensive logic during build configuration"?

Reply:
> My interpretation of this subtopic is that it's all about performance. My issue is that one could have lots of "logic" and configuration still be fast.

It's also about evaluation order and code maintainability I'd say. Though the main focus is performance.

> How about "Avoiding expensive logic during build configuration"?

I am OK with the suggestion.